### PR TITLE
RDM-6151-added chart and build pipeline

### PR DESCRIPTION
**Pull request by akdasari**
**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x ] No
```

### CODE REVIEW

File: ci-values.yaml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+global:
```
Comment (timja) [inline ci-values.yaml]: these global vars look unneeded in this chart? as it doesn't need any secrets

File: charts/ccd-docker-user-profile-importer/templates/_helpers.tpl [hunk 2]
```diff
@@ -0,0 +1,23 @@
+{{- define "importer.definition.vault" }}
```
Comment (timja) [inline charts/ccd-docker-user-profile-importer/templates/_helpers.tpl]: these global vars look unneeded in this chart? as it doesn't need any secrets

File: charts/ccd-docker-user-profile-importer/templates/user-profile-importer.yaml [hunk 3]
```diff
@@ -0,0 +1,63 @@
+{{- if .Values.importer.userprofile.enabled }}
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: "{{ .Release.Name }}-user-profile-importer"
```
Comment (adusumillipraveen) [inline charts/ccd-docker-user-profile-importer/templates/user-profile-importer.yaml]: Make releasename templated with releasenameoverride and tpl support 
https://github.com/hmcts/chart-java/blob/master/java/templates/deployment.yaml#L5
https://github.com/hmcts/chart-java/blob/master/java/templates/_helpers.tpl#L83

**Review by adusumillipraveen — CHANGES_REQUESTED** [review]

File: Makefile [hunk 4]
```diff
@@ -0,0 +1,36 @@
+.DEFAULT_GOAL := all
+CHART := ccd-user-profile-importer
+RELEASE := chart-${CHART}-release
+NAMESPACE := chart-tests
+TEST := ${RELEASE}-test-service
```
Comment (adusumillipraveen) [inline Makefile]: There are no tests ? May be move the tests from where this is copied? hoping it has some

File: charts/ccd-docker-user-profile-importer/templates/_helpers.tpl [hunk 5]
```diff
@@ -0,0 +1,14 @@
+{{- define "importer.definition.resourcegroup" }}
```
Comment (adusumillipraveen) [inline charts/ccd-docker-user-profile-importer/templates/_helpers.tpl]: Is this used?

File: charts/ccd-docker-user-profile-importer/templates/user-profile-importer.yaml [hunk 6]
```diff
@@ -0,0 +1,63 @@
+{{- if .Values.importer.userprofile.enabled }}
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: {{ template "hmcts.ccdupi.releaseName" . }}
+  labels:
+    app.kubernetes.io/name: {{ template "hmcts.ccdupi.releaseName" . }}
+    app.kubernetes.io/managed-by: {{ .Release.Service | quote }}
+    app.kubernetes.io/instance: {{ .Release.Name | quote }}
```
Comment (adusumillipraveen) [inline charts/ccd-docker-user-profile-importer/templates/user-profile-importer.yaml]: change this to template as well ?

File: charts/ccd-docker-user-profile-importer/templates/user-profile-importer.yaml [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+    helm.sh/hook: post-install,post-upgrade
+    helm.sh/hook-weight: "5"
+    helm.sh/hook-delete-policy: before-hook-creation  # Move to hook-succeeded after first charts are done 
+spec:
+  template:
+    metadata:
+      name: {{ template "hmcts.ccdupi.releaseName" . }}
+      labels:
+        app.kubernetes.io/managed-by: {{ .Release.Service | quote }}
+        app.kubernetes.io/instance: {{ .Release.Name | quote }}
```
Comment (adusumillipraveen) [inline charts/ccd-docker-user-profile-importer/templates/user-profile-importer.yaml]: same here?

File: charts/ccd-docker-user-profile-importer/templates/user-profile-importer.yaml [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+        helm.sh/chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
+    spec:
+      containers:
+        - name: {{ template "hmcts.ccdupi.releaseName" . }}
+          image: {{ .Values.importer.userprofile.image }} 
+          imagePullPolicy: IfNotPresent
+          env:
+          - name: CCD_USER_PROFILE_URL
+            value: http://{{ .Release.Name }}-user-profile-api
```
Comment (adusumillipraveen) [inline charts/ccd-docker-user-profile-importer/templates/user-profile-importer.yaml]: this should be externally configurable like .Values.useprofileUrl may be with tpl enabled .. doesn't make sense to be releasename dependent

File: charts/ccd-docker-user-profile-importer/values.yaml [hunk 9]
```diff
@@ -0,0 +1,18 @@
...
+    cpuRequests: '250m'
+    memoryLimits: '2048Mi'
+    cpuLimits: '1500m'
+    users:
+    jurisdictions:
+    waitHosts: ""
+    waitHostsTimeout: 300
+    microservice: ccd_definition
+    verbose: "false"
+    userProfileDatabaseHost: "${SERVICE_NAME}-ccd-postgres"
```
Comment (adusumillipraveen) [inline charts/ccd-docker-user-profile-importer/values.yaml]: these should be revisited?

File: charts/ccd-docker-user-profile-importer/templates/_helpers.tpl (hunk 5 above)
Comment (akdasari) [reply charts/ccd-docker-user-profile-importer/templates/_helpers.tpl]: not required as i removed ref in ci-values

File: Makefile [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+
+setup:
+	az account set --subscription ${ACR_SUBSCRIPTION}
+	az configure --defaults acr=${ACR}
+	az acr helm repo add --name ${HELM_REPO}
+	az aks get-credentials --resource-group ${AKS_RESOURCE_GROUP} --name ${AKS_CLUSTER} --overwrite-existing
+
+clean:
+	-helm delete --purge ${RELEASE}
+	-kubectl delete pod ${TEST} -n ${NAMESPACE}
```
Comment (adusumillipraveen) [inline Makefile]: not needed ?

File: charts/ccd-docker-user-profile-importer/Chart.yaml [hunk 11]
```diff
@@ -0,0 +1,7 @@
+description: Helm chart for the HMCTS CCD USER PROFILE Importer
+name: ccd-docker-user-profile-importer
```
Comment (adusumillipraveen) [inline charts/ccd-docker-user-profile-importer/Chart.yaml]:
```suggestion
name: ccd-user-profile-importer
```

File: charts/ccd-docker-user-profile-importer/values.yaml (hunk 9 above)
Comment (adusumillipraveen) [reply charts/ccd-docker-user-profile-importer/values.yaml]: can you please address this ? ${} notation might not work . Also do we need 
importer:
  userProfile: at root level. It made sense earlier when it was a big chart.

**Review by adusumillipraveen — CHANGES_REQUESTED** [review]

File: charts/ccd-docker-user-profile-importer/Chart.yaml (hunk 11 above)
Comment (akdasari) [reply charts/ccd-docker-user-profile-importer/Chart.yaml]: cleaned

File: Makefile (hunk 10 above)
Comment (akdasari) [reply Makefile]: cleaned

File: charts/ccd-docker-user-profile-importer/templates/user-profile-importer.yaml (hunk 8 above)
Comment (akdasari) [reply charts/ccd-docker-user-profile-importer/templates/user-profile-importer.yaml]: fixed

File: charts/ccd-docker-user-profile-importer/values.yaml [hunk 12]
```diff
@@ -0,0 +1,19 @@
+importer:
```
Comment (adusumillipraveen) [inline charts/ccd-docker-user-profile-importer/values.yaml]: can we change these to root level as this is not a sub chart ?

Comment (akdasari) [reply charts/ccd-docker-user-profile-importer/values.yaml]: fixed

File: charts/ccd-docker-user-profile-importer/values.yaml (hunk 9 above)
Comment (akdasari) [reply charts/ccd-docker-user-profile-importer/values.yaml]: fixed

File: charts/ccd-docker-user-profile-importer/values.yaml [hunk 13]
```diff
@@ -0,0 +1,17 @@
...
+memoryLimits: '2048Mi'
+cpuLimits: '1500m'
+userProfileUrl: 
+users:
+jurisdictions:
+waitHosts: ""
+waitHostsTimeout: 300
+microservice: ccd_definition
+verbose: "false"
+userProfileDatabaseHost: ccd-user-profile-api-postgres-db-{{ .Values.global.environment }}.postgres.database.azure.com
```
Comment (adusumillipraveen) [inline charts/ccd-docker-user-profile-importer/values.yaml]: doesnt make sense to point to Azure PAS DB when passwords are pointing to local postgresql

Comment (akdasari) [reply charts/ccd-docker-user-profile-importer/values.yaml]: modified the value now

**Review by adusumillipraveen — APPROVED** [review]
Looks ok to me

**Pull request merged**